src/Microsoft.Health.Dicom.Api/Controllers/WorkitemController.Update.cs

```diff
@@ -0,0 +1,77 @@
+
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Api.Features.Audit;
+using Microsoft.Health.Dicom.Api.Extensions;
+using Microsoft.Health.Dicom.Api.Features.Routing;
+using Microsoft.Health.Dicom.Core.Extensions;
+using Microsoft.Health.Dicom.Core.Features.Audit;
+using Microsoft.Health.Dicom.Core.Messages.Workitem;
+using Microsoft.Health.Dicom.Core.Models;
+using Microsoft.Health.Dicom.Core.Web;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.Health.Dicom.Api.Controllers;
+
+public partial class WorkitemController
+{
+    /// <summary>
+    /// This action requests the update of a UPS Instance on the Origin-Server. It corresponds to the UPS DIMSE N-SET operation.
+    /// </summary>
+    /// <remarks>
+    /// The request body contains all the metadata to be updated in DICOM PS 3.18 JSON metadata.
+    /// </remarks>
+    /// <returns></returns>
+    [HttpPost]
+    [Produces(KnownContentTypes.ApplicationDicomJson)]
+    [Consumes(KnownContentTypes.ApplicationDicomJson)]
+    [ProducesResponseType((int)HttpStatusCode.OK)]
+    [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+    [ProducesResponseType((int)HttpStatusCode.Conflict)]
+    [ProducesResponseType((int)HttpStatusCode.NotAcceptable)]
+    [ProducesResponseType((int)HttpStatusCode.NotFound)]
+    [ProducesResponseType((int)HttpStatusCode.UnsupportedMediaType)]
+    [VersionedPartitionRoute(KnownRoutes.UpdateWorkitemInstancesRoute, Name = KnownRouteNames.PartitionedUpdateWorkitemInstance)]
+    [VersionedRoute(KnownRoutes.UpdateWorkitemInstancesRoute, Name = KnownRouteNames.UpdateWorkitemInstance)]
+    [AuditEventType(AuditEventSubType.UpdateWorkitem)]
+    public async Task<IActionResult> UpdateAsync(string workitemInstanceUid)
+    {
+        // The Transaction UID is passed as the first query parameter 
+        string transactionUid = HttpContext.Request.Query.Keys.FirstOrDefault();
+
+        return await PostUpdateAsync(workitemInstanceUid, transactionUid);
+    }
+
+    private async Task<IActionResult> PostUpdateAsync(string workitemInstanceUid, string transactionUid)
+    {
+        long fileSize = Request.ContentLength ?? 0;
+        _logger.LogInformation("DICOM Web Update Workitem Transaction request received with file size of {FileSize} bytes.", fileSize);
+
+        UpdateWorkitemResponse response = await _mediator.UpdateWorkitemAsync(
+            Request.Body,
+            Request.ContentType,
+            workitemInstanceUid,
+            transactionUid,
+            HttpContext.RequestAborted);
+
+        if (response.Status == WorkitemResponseStatus.Success)
+        {
+            Response.Headers.Add(HeaderNames.ContentLocation, response.Uri.ToString());
+            Response.Headers.Add(HeaderNames.Location, response.Uri.ToString());
```

Comment:
I don't think we need Location - it's specified for Create only

Reply:
Spec says "Shall be present if the response has a payload containing a resource. See [Section 8.4.3](https://dicom.nema.org/medical/dicom/current/output/html/part18.html#sect_8.4.3).
May be present otherwise".